sources/api/migration/migrations/v1.11.0/aws-creds/src/main.rs

```diff
@@ -0,0 +1,35 @@
+#![deny(rust_2018_idioms)]
+
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added new settings for AWS credential configuration.
+fn run() -> Result<()> {
+    if cfg!(variant_platform = "aws") {
+        migrate(AddSettingsMigration(&[
+            "settings.aws.config",
+            "settings.aws.credentials",
+            "settings.aws.profile",
+        ]))
+    } else {
+        // Non-AWS variants did not have the region setting until this point,
+        // so need to be included here to remove on downgrade.
+        migrate(AddSettingsMigration(&[
+            "settings.aws.config",
+            "settings.aws.credentials",
+            "settings.aws.profile",
+            "settings.aws.region",
```

Comment:
I'm confused, I didn't see any update to the other platforms' models that added `AwsSettings` as in [AWS variants] (I also looked at the original PR but couldn't find anything there either). Is that going to happen in a different PR?

[AWS variants]: https://github.com/bottlerocket-os/bottlerocket/blob/ea2b374ea67ad634d20a18f83d3d4dfdb57bdcb1/sources/models/src/aws-k8s-1.24-nvidia/mod.rs#L25

Reply:
`AwsSettings` already was there, but it would only be present for `aws-*` variants. 

See Ben's comment here: https://github.com/bottlerocket-os/bottlerocket/pull/2377/files#r1010049045

Reply:
Those changes were being done in a later commit when they were actually being used for these other non-AWS variants. But to make it more clear as far as the migration is concerned I have now pulled those additions into this PR. So they will be available to the other variants, but a later PR will actually wire them in to be used when the other settings are configured.